### ISSUE TITLE
dfsioe yarn error

### ISSUE THREAD

**Issue by atupal** [intel-hadoop/HiBench#27]
when the dfsioe/run-write.sh
<br> it failed with the following error:
```

13/09/09 15:16:03 INFO mapreduce.Job: Task Id : attempt_1378468570219_0008_m_000000_2, Status : FAILED
Error: java.io.IOException: Mkdirs failed to create /benchmarks/TestDFSIO-Enh/io_data
        at org.apache.hadoop.fs.ChecksumFileSystem.create(ChecksumFileSystem.java:434)
        at org.apache.hadoop.fs.ChecksumFileSystem.create(ChecksumFileSystem.java:420)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:840)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:821)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:783)
        at org.apache.hadoop.fs.dfsioe.TestDFSIOEnh$WriteMapperEnh.doIO(TestDFSIOEnh.java:209)
        at org.apache.hadoop.fs.dfsioe.IOMapperBase.map(IOMapperBase.java:123)
        at org.apache.hadoop.fs.dfsioe.IOMapperBase.map(IOMapperBase.java:41)
        at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:54)
        at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:399)
        at org.apache.hadoop.mapred.MapTask.run(MapTask.java:334)
        at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:152)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:396)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1375)
        at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:147)
```
But the  /benchmarks/TestDFSIO-Enh/io_input have beed created successfuly,

**Comment by viplav** [comment]
Hi, 
We have been seeing the same issue with Hortonworks HDP 2 (at least since HDP 2.2). We really like dfsioe compared to standard Hadoop DFSIO, and I found some time to debug the issue. Tested a fix. Should I send a Pull request so that others who want to use dfsioe with HDP can benefit? Thanks.

**Comment by viplav** [comment]
I have sent a pull request ( https://github.com/intel-hadoop/HiBench/pull/110 ) to fix the issue with using wrong FS resulting in errors like  "Mkdirs failed to create" etc. Please take a look and merge so that people who run into this issue (e.g. using Hortonworks HDP 2) will benefit.

**Comment by somerandomsequence** [comment]
I'm still getting this error with the current master branch. Has it been merged in to master?

**Comment by adrian-wang** [comment]
cc @carsonwang

**Comment by carsonwang** [comment]
The fix mentioned in this issue was included in the master branch. We also fixed another issue in DFSIOe #276 . Please check the command that submits the application and what is the value of `test.build.data`.

**Comment by kumarshirish** [comment]
I get the following error while running dfsioe:
17/05/16 16:30:11 INFO dfsioe.TestDFSIOEnh: creating control file: 1 mega bytes, 16 files
17/05/16 16:30:11 INFO dfsioe.DfsioeConfig: Found 'test.build.data' in System properties, value is 'hdfs://nameservice1:8020/HiBench/Dfsioe/Input'
17/05/16 16:30:11 INFO dfsioe.DfsioeConfig: Setting testRootDir to 'hdfs://nameservice1:8020/HiBench/Dfsioe/Input'
java.lang.IllegalArgumentException: Wrong FS: hdfs://nameservice1:8020/HiBench/Dfsioe/Input/io_control, expected: hdfs://nameservice1
        at org.apache.hadoop.fs.FileSystem.checkPath(FileSystem.java:657)
        at org.apache.hadoop.hdfs.DistributedFileSystem.getPathName(DistributedFileSystem.java:194)
        at org.apache.hadoop.hdfs.DistributedFileSystem.access$000(DistributedFileSystem.java:106)
        at org.apache.hadoop.hdfs.DistributedFileSystem$12.doCall(DistributedFileSystem.java:646)
        at org.apache.hadoop.hdfs.DistributedFileSystem$12.doCall(DistributedFileSystem.java:642)
        at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
        at org.apache.hadoop.hdfs.DistributedFileSystem.delete(DistributedFileSystem.java:642)
        at org.apache.hadoop.fs.dfsioe.TestDFSIOEnh.createControlFile(TestDFSIOEnh.java:640)
        at org.apache.hadoop.fs.dfsioe.TestDFSIOEnh.run(TestDFSIOEnh.java:602)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:70)
        at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:84)
        at org.apache.hadoop.fs.dfsioe.TestDFSIOEnh.main(TestDFSIOEnh.java:628)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.hadoop.util.RunJar.run(RunJar.java:221)
        at org.apache.hadoop.util.RunJar.main(RunJar.java:136)

Here is the output on the console:
patching args=
Parsing conf: /var/lib/hadoop-hdfs/HiBench/conf/99-user_defined_properties.conf
Parsing conf: /var/lib/hadoop-hdfs/HiBench/conf/hadoop.conf
Parsing conf: /var/lib/hadoop-hdfs/HiBench/conf/hibench.conf
Parsing conf: /var/lib/hadoop-hdfs/HiBench/conf/spark.conf
Parsing conf: /var/lib/hadoop-hdfs/HiBench/conf/workloads/micro/dfsioe.conf
probe sleep jar: /opt/cloudera/parcels/CDH/lib/hadoop/../../jars/hadoop-mapreduce-client-jobclient-2.6.0-cdh5.8.0-tests.jar
start HadoopDfsioe-write bench
/var/lib/hadoop-hdfs/HiBench/bin/functions/workload_functions.sh: line 115: /dev/stderr: Permission denied
rm: `hdfs://nameservice1:8020/HiBench/Dfsioe/Output': No such file or directory
/var/lib/hadoop-hdfs/HiBench/bin/functions/workload_functions.sh: line 152: /dev/stderr: Permission denied
du: `hdfs://nameservice1:8020/HiBench/Dfsioe/Input': No such file or directory
Submit MapReduce Job: /opt/cloudera/parcels/CDH/lib/hadoop/bin/hadoop --config /opt/cloudera/parcels/CDH/lib/hadoop/etc/hadoop jar /var/lib/hadoop-hdfs/HiBench/autogen/target/autogen-6.1-SNAPSHOT-jar-with-dependencies.jar org.apache.hadoop.fs.dfsioe.TestDFSIOEnh -Dmapreduce.map.java.opts="-Dtest.build.data=hdfs://nameservice1:8020/HiBench/Dfsioe/Input -Djava.net.preferIPv4Stack=true" -Dmapreduce.reduce.java.opts="-Dtest.build.data=hdfs://nameservice1:8020/HiBench/Dfsioe/Input -Djava.net.preferIPv4Stack=true" -write -nrFiles 16 -fileSize 1 -bufferSize 4096 -plotInteval 1000 -sampleUnit m -sampleInteval 200 -sumThreshold 0.5 -tputReportTotal -Dtest.build.data=hdfs://nameservice1:8020/HiBench/Dfsioe/Input -resFile /var/lib/hadoop-hdfs/HiBench/report/dfsioe/hadoop/conf/../result_write.txt -tputFile /var/lib/hadoop-hdfs/HiBench/report/dfsioe/hadoop/conf/../throughput_write.csv
        at org.apache.hadoop.util.RunJar.main(RunJar.java:136)
ERROR: Hadoop job /var/lib/hadoop-hdfs/HiBench/autogen/target/autogen-6.1-SNAPSHOT-jar-with-dependencies.jar org.apache.hadoop.fs.dfsioe.TestDFSIOEnh failed to run successfully.
Hint: You can goto /var/lib/hadoop-hdfs/HiBench/report/dfsioe/hadoop/conf/../bench.log to check for detailed log.
Opening log tail for you:

	at org.apache.hadoop.fs.dfsioe.TestDFSIOEnh.run(TestDFSIOEnh.java:602)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:70)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:84)
	at org.apache.hadoop.fs.dfsioe.TestDFSIOEnh.main(TestDFSIOEnh.java:628)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:221)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:136)

This happens whether while running prepare, read, or run_all.sh.
